homeassistant/components/wilight/__init__.py

```diff
@@ -0,0 +1,308 @@
+"""The WiLight integration."""
+import asyncio
+import logging
+
+import pywilight
+import requests
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, DT_CONFIG, DT_PENDING, DT_REGISTRY, DT_SERIAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def coerce_host_port(value):
+    """Validate that provided value is either just host or host:port.
+
+    Returns (host, None) or (host, port) respectively.
+    """
+    host, _, port = value.partition(":")
+
+    if not host:
+        raise vol.Invalid("host cannot be empty")
+
+    if port:
+        port = cv.port(port)
+    else:
+        port = None
+
+    return host, port
+
+
+CONF_STATIC = "static"
+CONF_DISCOVERY = "discovery"
+
+DEFAULT_DISCOVERY = True
+
+CONFIG_SCHEMA = vol.Schema(
```

Comment:
Please remove the YAML configuration. Integration that communicates with a device or service, should be using the configuration flow only.

Reply:
See: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>